Prepare multi-arch support

steveworley opened the pull request:
Prepare multi-arch support for better compatibility with Apple M1 (arm64) processors.

- Test the publish stage

steveworley committed:
Prepare multiarch support.

Update bake.hcl and deploy for multiarch.

Add multiarch docker context.

Add builder for docker buildx.

Set default docker context.

Add image tag to build.

Add builder arg for buildx.

Change global docker context.

Update build context.

Enable docker experimental.

Change context creation.

Remove the build as default context is set.

Update the php image for builder tests.

Update to test image tags.

Add image tag var

Update tests to match container names.

Set IMAGE_TAG for whole build.

Update image tag for build.

Fix typo

Add DOCKERHUB_NAMESPACE.

Move test into single step.

Fix typo

steveworley committed:
Add push option to publish.

steveworley committed:
Update README.md

steveworley committed:
Rejig the CI pipeline.

christopher-hopper commented on .circleci/config.yml:
```diff
@@ -3,50 +3,76 @@ version: 2
+          command: |
+            docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+            docker context create dpc
+            docker buildx create --platform linux/amd64,linux/arm64 dpc --name dpc --use
```
My Apple M1  says its detected platform is `linux/arm/v8` so maybe need to build for that too? like:

```
docker buildx create --platform linux/amd64,linux/arm64,linux/arm/v8 
```

christopher-hopper committed:
Merge branch '4.x' into build/multi-arch

christopher-hopper commented:
@steveworley 
It looks like changes from this branch were merged. Assume this PR can be closed?

christopher-hopper committed:
Revert README.md changes

christopher-hopper committed:
Use mariadb-10.4 base image with arm64 build

This versioned image matches the unversioned base image but has arm64
compatible build.

christopher-hopper commented:
@steveworley @GROwen 
I'm going to continue work in this branch to further progress multi-arch support.

christopher-hopper commented:
MariaDB build works and new arm64 image is being built from the arm64 compatible base. You can see that it is slightly smaller than the amd64 image, which is as expected. 

<img width="1055" alt="Screen Shot 2022-05-03 at 4 04 55 pm" src="https://user-images.githubusercontent.com/452515/166410946-ae2295d1-58a4-4cf0-b0cf-21ded1fd99fb.png">


The image however has no customisations from the base uselagoon/mariadb-10.4-drupal image. As such, you could really stop building and publishing this image altogether.

GROwen commented:
Closing in favour of https://github.com/dpc-sdp/bay/pull/108
